src/functions/Set-ItResult.ps1

```diff
@@ -67,60 +54,14 @@ function Set-ItResult {
     Assert-DescribeInProgress -CommandName Set-ItResult
 
     $result = $PSCmdlet.ParameterSetName
-    $message = "It result set to $result$(if ($Because) { ", $Because" })"
-    $data = @{
-        Result  = $result
-        Because = $Because
-    }
-    $errorRecord = New-PesterErrorRecord -Result $result  -Message $message -Data $data
```

Comment:
this was imho okay, you'd just change the message here, and change the Pester.Factory method to also accept data, and that should be it, throwing a runtime exception makes this a possbile conflict with an exception that a user would throw, and we don't want that. (Sure they can throw our exception type, but that is was more unlikely.)

Reply:
It now throws using the Pester.Factory method